### PR TITLE
Use pj in ttj, tj, tej, tfj and 'tej enum' ##refactor

### PR DESCRIPTION
**Pull request by a-meth**
Used pj.c api for ttj

### CODE REVIEW

File: libr/core/cmd_type.c [hunk 1]
```diff
@@ -274,19 +274,37 @@ static int printkey_json_cb(void *user, const char *k, const char *v) {
 
 static void printFunctionType(RCore *core, const char *input) {
 	Sdb *TDB = core->anal->sdb_types;
+	PJ *pj = NULL;
```
Comment (radare) [inline libr/core/cmd_type.c]: assign and define in the same line

File: libr/core/cmd_type.c [hunk 2]
```diff
@@ -274,19 +274,37 @@ static int printkey_json_cb(void *user, const char *k, const char *v) {
...
 		char *name = strchr (type, ',');
 		if (name) {
 			*name++ = 0;
 		}
-		r_cons_printf ("{\"type\":\"%s\",\"name\":\"%s\"}%s", type, name, i+1==args? "": ",");
+		pj_o (pj);
+		pj_ks (pj, "type", type);
+		if (name != NULL) {
+			pj_ks (pj, "name", name);
+		}
```
Comment (radare) [inline libr/core/cmd_type.c]: else in the same line as the }

File: libr/core/cmd_type.c [hunk 3]
```diff
@@ -274,19 +274,37 @@ static int printkey_json_cb(void *user, const char *k, const char *v) {
...
 	for (i = 0; i< args; i++) {
 		char *type = sdb_get (TDB, sdb_fmt ("func.%s.arg.%d", name, i), 0);
 		char *name = strchr (type, ',');
 		if (name) {
 			*name++ = 0;
 		}
-		r_cons_printf ("{\"type\":\"%s\",\"name\":\"%s\"}%s", type, name, i+1==args? "": ",");
+		pj_o (pj);
+		pj_ks (pj, "type", type);
+		if (name != NULL) {
```
Comment (radare) [inline libr/core/cmd_type.c]: != NULL is unnecessary

File: libr/core/cmd_type.c [hunk 4]
```diff
@@ -343,14 +361,22 @@ static int print_typelist_r_cb(void *p, const char *k, const char *v) {
 
 static int print_typelist_json_cb(void *p, const char *k, const char *v) {
 	RCore *core = (RCore *)p;
+	PJ *pj = NULL;
```
Comment (radare) [inline libr/core/cmd_type.c]: define and assign in the same line

File: libr/core/cmd_type.c [hunk 5]
```diff
@@ -343,14 +361,22 @@ static int print_typelist_r_cb(void *p, const char *k, const char *v) {
...
 	char *formatcmd = r_str_newf ("type.%s", k);
 	char *format_s = r_str_trim (sdb_querys (sdb, NULL, -1, formatcmd));
-	r_cons_printf ("{\"type\":\"%s\",\"size\":%d,\"format\":\"%s\"}", k,
-			size_s ? atoi (size_s) : -1,
-			format_s);
+	pj_ks (pj, "type", k);
+	pj_ki (pj, "size", size_s ? atoi (size_s) : -1);
+	pj_ks (pj, "format", format_s);
+	pj_end (pj);
+	if (pj) {
```
Comment (radare) [inline libr/core/cmd_type.c]: pj should not be null here

File: libr/core/cmd_type.c [hunk 6]
```diff
@@ -821,53 +847,62 @@ static int cmd_type(void *data, const char *input) {
 				SdbKv *kv;
 				SdbListIter *iter;
 				SdbList *l = sdb_foreach_list (TDB, true);
-				const char *comma = "";
-				r_cons_printf ("{");
+				PJ *pj = NULL;
```
Comment (radare) [inline libr/core/cmd_type.c]: same h ere

Comment (radare) [reply libr/core/cmd_type.c]: chk  null

File: libr/core/cmd_type.c (hunk 5 above)
Comment (a-meth) [reply libr/core/cmd_type.c]: What?

File: libr/core/cmd_type.c (hunk 3 above)
Comment (a-meth) [reply libr/core/cmd_type.c]: WARNING: r_strbuf_append: assertion 'sb && s' failed (line 110)

File: libr/core/cmd_type.c (hunk 5 above)
Comment (a-meth) [reply libr/core/cmd_type.c]: You mean not to check pj and just print it directly?

File: libr/core/cmd_type.c [hunk 7]
```diff
@@ -274,19 +274,33 @@ static int printkey_json_cb(void *user, const char *k, const char *v) {
 
 static void printFunctionType(RCore *core, const char *input) {
 	Sdb *TDB = core->anal->sdb_types;
+	PJ *pj = pj_new ();
+	pj_o (pj);
```
Comment (radare) [inline libr/core/cmd_type.c]: Nullchk here

File: libr/core/cmd_type.c (hunk 3 above)
Comment (radare) [reply libr/core/cmd_type.c]: Wat. Just remoce the != NULL. If (name) is enough

File: libr/core/cmd_type.c (hunk 7 above)
Comment (a-meth) [reply libr/core/cmd_type.c]: What? Didn't understand what you meant?

Comment (radare) [reply libr/core/cmd_type.c]: see  my commit

**Pull request merged**